tests/unit/tuner/test_evaluation.py

```diff
@@ -0,0 +1,164 @@
+import math
+import pytest
+
+from jina import Document
+from finetuner import __default_tag_key__
+from finetuner.tuner.evaluation import _doc_to_relevance, Evaluator
+
+
+doc1 = Document(id="1")
+doc2 = Document(id="2")
+doc3 = Document(id="3")
+doc4 = Document(id="4")
+
+# evaluation docs
+doc5 = Document(
+    matches=[doc1, doc2, doc3, doc4],
+    tags={__default_tag_key__: {"targets": {"1": 3, "3": 1}}},
+)
+doc6 = Document(
+    matches=[doc1, doc4], tags={__default_tag_key__: {"targets": {"4": 3, "1": 1}}}
+)
+
+# doc with no matches
+doc7 = Document(matches=[], tags={__default_tag_key__: {"targets": {"1": 3, "3": 1}}})
+
+# doc with no ground truth
+doc8 = Document(
+    matches=[doc1, doc2, doc3, doc4], tags={__default_tag_key__: {"targets": {}}}
+)
+
+# doc with no matches and no ground truth
+doc9 = Document(matches=[], tags={__default_tag_key__: {"targets": {}}})
```

Comment:
This should all be fixtures. Otherwise, one test changing one of the docs would have side-effect towards other tests. Having them as fixtures makes sure, tests are completely decoupled. E.g.
```
@pytest.fixture
def doc5():
    return Document(matches=[Document(id="1")]...)

def test_r_precision(doc5, ...):
    ...
```

Reply:
@maximilianwerk check out the new tests with the fixtures

Reply:
Much better! Thanks.